activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskUpdater.java

```diff
@@ -0,0 +1,122 @@
+        this.taskService = taskService;
+        this.securityManager=securityManager;
+        this.userGroupManager = userGroupManager;
+        this.taskConverter = taskConverter;
+    }
+    
+    public Task applyUpdateTaskPayload(boolean isAdmin, UpdateTaskPayload updateTaskPayload) {
+        
+        org.activiti.engine.task.Task internalTask=null;
+        int updates=0;
+        String oldValue,newValue;
+        
+        if (isAdmin) {
+            internalTask = taskService.createTaskQuery().taskId(updateTaskPayload.getTaskId()).singleResult();
+        } else {
+            internalTask=getInternalTaskWithChecks(updateTaskPayload.getTaskId());
+        }
+        
+        if (internalTask == null) {
+            throw new NotFoundException("Unable to find task for the given id: " + updateTaskPayload.getTaskId());
+        }
+          
+        if ((newValue = updateTaskPayload.getName()) != null) {
+            oldValue = internalTask.getName();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setName(newValue);
+            }
+        }
+        
+        if ((newValue = updateTaskPayload.getDescription()) != null) {
+            oldValue = internalTask.getDescription();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setDescription(newValue);
+            }
+        }
+            
+        if (updateTaskPayload.getPriority() != null) {
+            if (internalTask.getPriority()!=updateTaskPayload.getPriority()) {
+                updates++;
+                internalTask.setPriority(updateTaskPayload.getPriority());
+            }
+        }
+        
+        if (updateTaskPayload.getDueDate() != null) {
+            if (!Objects.equals(internalTask.getDueDate(),updateTaskPayload.getDueDate())) {
+                updates++;
+                internalTask.setDueDate(updateTaskPayload.getDueDate());
+            }
+        }
+        
+        //@TODO: check if this value can be updated
+        if ((newValue=updateTaskPayload.getParentTaskId()) != null) {
+            oldValue = internalTask.getParentTaskId();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setParentTaskId(newValue);
+            }
+        }
+        
+        if ((newValue=updateTaskPayload.getFormKey()) != null) {
+            oldValue = internalTask.getFormKey();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setFormKey(newValue);
+            }
+        }
+        
+        if (updates > 0) {
+            taskService.saveTask(internalTask);
+        }
+        
+        return taskConverter.from(taskService.createTaskQuery().taskId(updateTaskPayload.getTaskId()).singleResult());
+    }
+    
+    public org.activiti.engine.task.Task getInternalTaskWithChecks(String taskId) {
```

Comment:
Ok, we need to find a better name than `TaskUpdater` for this class, as we need remove the method `getInternalTaskWithChecks` from `TaskRuntimeImpl` and reuse this one (to avoid duplication). Maybe `TaskRuntimeHelper` while we wait for a better name (or anything else you think it makes sense).